qiskit_nature/transformers/active_space_transformer.py

```diff
@@ -0,0 +1,193 @@
+        if self.num_alpha is not None:
+            if not beta:
+                warning = 'The provided instance of QMolecule does not provide any beta ' \
+                          + 'coefficients but you tried to specify a separate number of alpha ' \
+                          + 'electrons. Continuing as if it does not matter.'
+                logger.warning(warning)
+            num_alpha = self.num_alpha
+            num_beta = self.num_electrons - self.num_alpha
+        else:
+            num_beta = (self.num_electrons - (q_molecule.multiplicity - 1)) // 2
+            num_alpha = self.num_electrons - num_beta
+        if nelec_inactive < 0:
+            raise QiskitNatureError("More electrons requested than available.")
+        if nelec_inactive % 2 != 0:
+            raise QiskitNatureError("The number of inactive electrons must be even.")
+
+        # determine active and inactive orbital indices
+        if self.active_orbitals is None:
+            norbs_inactive = nelec_inactive // 2
+            if norbs_inactive + self.num_orbitals > q_molecule.num_orbitals:
+                raise QiskitNatureError("More orbitals requested than available.")
+            inactive_orbs_idxs = list(range(norbs_inactive))
+            active_orbs_idxs = list(range(norbs_inactive, norbs_inactive+self.num_orbitals))
+        else:
+            if self.num_orbitals != len(self.active_orbitals):
+                raise QiskitNatureError("The number of selected active orbital indices does not "
+                                        "match the specified number of active orbitals.")
+            if max(self.active_orbitals) >= q_molecule.num_orbitals:
+                raise QiskitNatureError("More orbitals requested than available.")
+            active_orbs_idxs = self.active_orbitals
+            inactive_orbs_idxs = [o for o in range(nelec_total // 2) if o not in
+                                  self.active_orbitals and mo_occ_total[o] > 0]
+
+        # split molecular orbitals coefficients into active and inactive parts
+        mo_coeff_inactive = mo_coeff_full[:, inactive_orbs_idxs]
+        mo_coeff_active = mo_coeff_full[:, active_orbs_idxs]
+        mo_coeff_active_b = mo_coeff_inactive_b = None
+        if beta:
+            mo_coeff_inactive_b = mo_coeff_full_b[:, inactive_orbs_idxs]
+            mo_coeff_active_b = mo_coeff_full_b[:, active_orbs_idxs]
+
+        # compute inactive density matrix
+        mo_occ_inactive = mo_occ_full[inactive_orbs_idxs]
+        density_inactive = np.dot(mo_coeff_inactive*mo_occ_inactive,
+                                  np.transpose(mo_coeff_inactive))
+        density_inactive_b = None
+        if beta:
+            mo_occ_inactive_b = mo_occ_full_b[inactive_orbs_idxs]
+            density_inactive_b = np.dot(mo_coeff_inactive_b*mo_occ_inactive_b,
+                                        np.transpose(mo_coeff_inactive_b))
+
+        # compute inactive Fock matrix
```

Comment:
I would extract computing the Fock matrix to a helper function for improved readibility and compactness. Similarly for other fragments annotated with comments.